dali/operators/coord/coord_flip.cu

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/coord/coord_flip.h"
+
+namespace dali {
+
+namespace {
+
+template <typename T = float>
+struct SampleDesc {
+  float* out = nullptr;
+  const float* in = nullptr;
+  int64_t size = 0;
+  uint8_t flip_dim_mask = 0;
+};
+
+template <typename T = float>
+__global__ void CoordFlipKernel(const SampleDesc<T>* samples, int ndim) {
+  int64_t block_size = blockDim.y * blockDim.x;
+  int64_t grid_size = gridDim.x * block_size;
+  int sample_idx = blockIdx.y;
+  const auto &sample = samples[sample_idx];
+  int64_t offset = block_size * blockIdx.x;
+  int64_t tid = threadIdx.y * blockDim.x + threadIdx.x;
+  for (int64_t idx = offset + tid; idx < sample.size; idx += grid_size) {
+    int d = idx % ndim;
+    bool flip = static_cast<bool>(sample.flip_dim_mask & (1 << d));
```

Comment:
I think implicit conversion would do just fine, like it would in `if` statement.

Reply:
done